### PR TITLE
fix(nested row): border color and align fixed

### PR DESCRIPTION
**Pull request by jagathgj**
Contributes to #2505 

Design review fixes done for the nested rows. Alignment of columns fixed as well as border color changed

#### What did you change?
scss changed and also condition for getCellProps and geRowStyles in useNestedRows changed

#### How did you test and verify your work?
storybook

### CODE REVIEW

**Comment by davidmenendez** [comment]
tests are failing

**Review by davidmenendez — DISMISSED** [review]

File: packages/cloud-cognitive/src/components/Datagrid/styles/_useNestedRows.scss [hunk 1]
```diff
@@ -46,3 +46,20 @@
...
+
+.#{$block-class} tr.#{$block-class}__carbon-nested-row {
+  td:first-child:empty,
+  .#{$block-class}__expandable-row-cell {
+    border-bottom: none;
+  }
+}
+
+.#{$carbon-prefix}--data-table td.#{$block-class}__expandable-row-cell {
+  padding-left: 0;
```
Comment (elycheea) [inline packages/cloud-cognitive/src/components/Datagrid/styles/_useNestedRows.scss:58]: I think this should actually use `$spacing-03`.

Using `padding-left: 0`, the button gets shifted a little to the left.

Collapsed | Expanded
-----------|----------
image | image

**Review by elycheea — CHANGES_REQUESTED** [review]
Noticed one small issue with the left padding but otherwise looks good!

File: packages/cloud-cognitive/src/components/Datagrid/styles/_useNestedRows.scss (hunk 1 above)
Comment (jagathgj) [reply packages/cloud-cognitive/src/components/Datagrid/styles/_useNestedRows.scss]: spacing added.

**Review by elycheea — DISMISSED** [review]

**Review by elycheea — APPROVED** [review]

**Pull request merged**